client/attest.go

```diff
@@ -24,6 +33,78 @@ type AttestOpts struct {
 	// firmware event log, where PCRs 0-9 and 14 are often measured. If the two
 	// logs overlap, server-side verification using this library may fail.
 	CanonicalEventLog []byte
+	// Indicates whether the AK certificate chain should be retrieved for validation.
+	// If true, Key.Attest() will construct the certificate chain by making GET requests to
+	// the contents of Key.cert.IssuingCertificateURL.
+	FetchCertChain bool
+}
+
+// Given a certificate, iterates through its IssuingCertificateURLs and returns the certificate
+// that signed it. Returns an error if a valid signing certificate is not found.
+func fetchIssuingCertificate(cert *x509.Certificate) (*x509.Certificate, error) {
+	if cert == nil {
+		return nil, fmt.Errorf("provided certificate is nil")
+	}
+	// For each URL, fetch and parse the certificate, then verify whether it signed cert.
+	// If successful, return the parsed certificate. If any step in this process fails, try the next url.
+	for _, url := range cert.IssuingCertificateURL {
+		resp, err := http.Get(url)
+		if err != nil {
+			log.Printf("failed to retrieve certificate at %v: %v\n", url, err)
+			continue
+		}
+		defer resp.Body.Close()
```

Comment:
I think we should just read the response body and close it inside the current loop iteration, instead of deferring it (because if defer to func end, only the last `resp.Body` can be closed).